iotedgedev/modules.py

```diff
@@ -15,6 +14,53 @@ def __init__(self, envvars, utility, output, dock):
         self.dock = dock
         self.dock.init_registry()
 
+    def create(self, name, template):
+        self.output.header("CREATING MODULE")
+
+        repo = "{0}/{1}".format(self.envvars.CONTAINER_REGISTRY_SERVER, name.lower())
+        cwd = self.envvars.MODULES_PATH
+        if template == "csharp":
+            cmd = "dotnet new -i Microsoft.Azure.IoT.Edge.Module"
+            self.output.header(cmd)
+            self.utility.exe_proc(cmd.split())
+            cmd = "dotnet new aziotedgemodule -n {0} -r {1}".format(name, repo)
+            self.output.header(cmd)
+            self.utility.exe_proc(cmd.split(), cwd=cwd)
+        elif template == "nodejs":
+            pass
+        elif template == "python":
+            github_source = "https://github.com/Azure/cookiecutter-azure-iot-edge-module"
+            branch = "master"
+            cmd = ("cookiecutter --no-input {0} module_name={1} image_repository={2} --checkout {3}"
+                   .format(github_source, name, repo, branch))
+            self.output.header(cmd)
+            self.utility.exe_proc(cmd.split(), cwd=cwd)
+        elif template == "csharpfunction":
+            cmd = "dotnet new -i Microsoft.Azure.IoT.Edge.Function"
+            self.output.header(cmd)
+            self.utility.exe_proc(cmd.split())
+            cmd = "dotnet new aziotedgefunction -n {0} -r {1}".format(name, repo)
+            self.output.header(cmd)
+            self.utility.exe_proc(cmd.split(), cwd=cwd)
+
+        new_module = """{
```

Comment:
I feel like the below code could be extracted into a new class called deploymentfile.py that has a method called add.

Then you can easily test it.

Reply:
Thanks. I have added a new `DeploymentManifest` module.